spec/services/group_stage_service_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+RSpec.describe GroupStageService do
+  before do
+    @stage = create(:stage)
+    @teams1 = create_list(:team, 4)
+    @teams2 = create_list(:team, 4)
+    @groups = Hash[1 => @teams1, 2 => @teams2].values
+  end
+  describe 'generate_group_stage method' do
+    it 'returns a stage object' do
+      group_stage = GroupStageService.generate_group_stage(@groups)
+      expect(group_stage).to be_a(Stage)
+    end
+
+    it 'returns a stage object with level -1' do
+      group_stage_level = GroupStageService.generate_group_stage(@groups).level
+      expect(group_stage_level).to be(-1)
+    end
+
+    it 'adds the provided groups to the stage' do
+      group_stage_teams = GroupStageService.generate_group_stage(@groups).teams
+      expect(group_stage_teams).to match_array(@groups.flatten)
+    end
+
+    it 'returns false when given different sizes of groups' do
+      unequal_groups = @groups
+      unequal_groups.first.pop
+      group_stage = GroupStageService.generate_group_stage(@groups)
+      expect(group_stage).to eq(false)
+    end
+  end
+
+  describe 'get_group_object_from' do
+    it 'returns a group' do
+      group = GroupStageService.get_group_object_from(@teams1)
+      expect(group).to be_a(Group)
+    end
+  end
+
+  describe 'generate_all_matches_between' do
```

Comment:
use `#<method name>` for referring to methods (to be consistent with other specs, see controller specs for examples)